vars/dockerExecute.groovy

```diff
@@ -1,51 +1,65 @@
 import com.cloudbees.groovy.cps.NonCPS
+import com.sap.piper.ConfigurationLoader
 
 def call(Map parameters = [:], body) {
 
     def STEP_NAME = 'dockerExecute'
     def PLUGIN_ID_DOCKER_WORKFLOW = 'docker-workflow'
 
-    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters){
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
         def dockerImage = parameters.dockerImage ?: ''
         Map dockerEnvVars = parameters.dockerEnvVars ?: [:]
         def dockerOptions = parameters.dockerOptions ?: ''
         Map dockerVolumeBind = parameters.dockerVolumeBind ?: [:]
+        final script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
+        if (env.POD_NAME && hasContainerDefined(script, dockerImage)) {
+            container(getContainerDefined(script, dockerImage)) {
+                echo "Executing inside a Kubernetes Container"
+                body()
+                sh "chown -R 1000:1000 ."
+            }
+        } else if (env.jaas_owner) {
```

Comment:
jaas_owner, maybe we find another environment variable name, which is not sap specific and set and in all jaas templates.

Reply:
Yes. This is under discussion. We should remove it.

Reply:
I would also be happy if we can find a way, without using the JaaS internal variable.

Reply:
in Our pipeline, we would use something like this 
```
general:
  productiveBranch: 'master'
  kubernetes:
    enabled: false
    jnlpAgent: 'jenkinsci/jnlp-slave:latest'
 ```

But for Jaas we need to find a way.